src/compiler/checker.ts

```diff
@@ -936,6 +937,25 @@ namespace ts {
             return false;
         }
 
+
+        function checkAndReportErrorForExtendingInterface(errorLocation: Node, name: string): boolean {
+            if (!errorLocation || errorLocation.kind !== SyntaxKind.Identifier ||
+                !errorLocation.parent || !errorLocation.parent.parent ||

```

Comment:
For such cases 
```typescript
namespace N {
  export interface I {}
}
class C extends N.I {}
```

`resolveEntityName` does not work here because it requires a `ExpressionWithTypeArguments` node, while `N.I` is binded as a `expression`. 

Do you have a suggestion?